Fixes & refactor

daycry opened the pull request.

daycry committed:
Fix: Explode arguments

daycry committed:
Update README.md

daycry committed:
Merge branch 'angelov:main' into main

daycry committed:
Fix: Get name test for recording

daycry committed:
Fix: php-vcr version

angelov commented:
Thank you for the contribution @daycry. Can you please provide more info why we need these changes?

angelov committed:
Allow PHPUnit 12 (#4)

angelov committed:
Upgrade PHPStan (#6)

Commit:
Fixes & refactor

daycry committed:
Merge branch 'main' of https://github.com/daycry/phpunit-php-vcr

daycry commented:
My changes imported from https://github.com/[daycry/phpunit-extension-vcr](https://github.com/daycry/phpunit-extension-vcr)

daycry commented:
Coding standards checks with composer cs command

angelov commented:
There are too many unrelated changes for a single PR. Please split them into several PRs instead, so we can discuss them separately. Don't forget to add description and examples on why the changes are needed.

Here are few things which I don't think are needed:
* Additional Composer dependencies, config files, and script aliases
* Badges in the readme file (already explained in #2)
* Git hooks - We already have the workflows here
* License info in each file - There's a license file

daycry commented:
Sorry, but I can't spend more time updating your package. 

You told me we can collaborate and I do. If you install want you can check these changes or you can delete it, but the changes made are very interesting.

angelov commented:
I asked you to push the changes that **you needed** here, so you can correct the fact that you had copied the code and removed the references to the original work. I can support adding these changes to this package, but that does not mean you can just throw a bunch of changes in one PR without any additional context. If what is present here does not work for you, you are free to fork it and modify it as you wish, but you have to respect the license.

daycry commented:
there are many changes in all the files because I have added your signature to the beginning of each file.
I have modified the cs-fixer package to keep it up to the standards, At the code level I have basically modified how you retrieve the method or class to get the attribute.

If you are scared to see so many changes, I will divide them.
